compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -57,6 +57,7 @@ pub fn walk_native_lib_search_dirs<R>(
     if sess.target.vendor == "fortanix"
         || sess.target.os == "linux"
         || sess.target.os == "fuchsia"
+        || sess.target.is_like_aix
```

Comment:
This does more than just handle `libunwind`. For us on AIX, this will also set `/usr/lib` in our loader sections. For example:

```
$ dump -H /home/hjiang/build/powerpc64-ibm-aix/test/run-make/codegen-options-parsing/rmake

/home/hjiang/build/powerpc64-ibm-aix/test/run-make/codegen-options-parsing/rmake:

			***Loader Section***
                      Loader Header Information
VERSION#         #SYMtableENT     #RELOCent        LENidSTR
0x00000001       0x000000a5       0x00003e52       0x000000dd       

#IMPfilID        OFFidSTR         LENstrTBL        OFFstrTBL
0x00000005       0x0003f4d0       0x000028e1       0x0003f5ad       

			***Import File Strings***
INDEX  PATH                          BASE                MEMBER              
0      /home/hjiang/build/powerpc64-ibm-aix/stage1/lib/rustlib/powerpc64-ibm-aix/lib:/usr/lib:/lib                                         
1                                    libc.a              shr_64.o            
2                                    libpthread.a        shr_xpg5_64.o       
3                                    libstd-d65e3a4d7e9edd92.a libstd-d65e3a4d7e9edd92.so
4                                    libunwind.a         libunwind.so.1 
```

Otherwise, `libc.a` will not be resolved correctly in the run-make suites, and other binaries.

Reply:
Added as requested